src/class-parsely.php

```diff
@@ -586,4 +589,54 @@ public function get_all_track_types(): array {
 	public function get_default_options() {
 		return $this->option_defaults;
 	}
+
+	/**
+	 * Injects any managed credentials into the passed plugin options.
+	 *
+	 * This allows hosting providers to provide a more customized experience for
+	 * the plugin by handling credentials automatically.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @param Parsely_Options $options The options without managed credentials.
+	 *
+	 * @return Parsely_Options The options including the managed credentials.
+	 */
+	private function inject_managed_credentials( $options ) {
+		$credentials = apply_filters( 'wp_parsely_credentials', false );
+
+		if ( ! is_array( $credentials ) ) {
+			return $options;
+		}
+
+		if ( isset( $credentials['site_id'] ) ) {
+			$options['apikey'] = $credentials['site_id'];
+		}
+
+		if ( isset( $credentials['api_secret'] ) ) {
+			$options['api_secret'] = $credentials['api_secret'];
+		}
+
+		if ( isset( $credentials['metadata_secret'] ) ) {
+			$options['metadata_secret'] = $credentials['metadata_secret'];
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Returns whether credentials are being managed at the platform level.
+	 *
+	 * This allows hosting providers to provide a more customized experience for
+	 * the plugin by handling credentials automatically.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @return bool Whether credentials are being managed at the platform level.
+	 */
+	public static function are_credentials_managed(): bool {
+		return has_filter( 'wp_parsely_credentials' );
```

Comment:
I don't think `has_filter` is safe for use here.

It might be better for `inject_managed_credentials` to add a value that we can check instead?

```suggestion
				$options = $this->get_options();
				if ( isset( $options['managed_credentials'] ) && true === $options['managed_credentials'] ) {
					return true;
				}

				return false;
```

return isset( $this->options{
```

Reply:
has_filter tells us if a callback was attached not if the values were changed. If the callback returns early, managed mode should not be activated.

Reply:
Although I'm not sure there's an issue here, I agree that an option will be more resilient and I'm happy to go that route. I've started an internal discussion about that.

Reply:
@mehmoodak, @mjangda, I've implemented something based on our discussion.